packages/constants/assets.ts

```diff
@@ -1,165 +1,148 @@
-import { Asset } from '@penumbra-zone/types';
+import { DenomMetadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1alpha1/asset_pb';
+import { base64ToUint8Array } from '@penumbra-zone/types';
 
-export interface AssetPattens {
-  lpNftPattern: RegExp;
-  delegationTokenPattern: RegExp;
-  proposalNftPattern: RegExp;
-  unbondingTokenPattern: RegExp;
-  votingReceiptPattern: RegExp;
+export interface AssetPatterns {
+  lpNft: RegExp;
+  delegationToken: RegExp;
+  proposalNft: RegExp;
+  unbondingToken: RegExp;
+  votingReceipt: RegExp;
 }
 
-export const assetPatterns: AssetPattens = {
-  lpNftPattern: new RegExp('^lpnft_'),
-  delegationTokenPattern: new RegExp('^delegation_'),
-  proposalNftPattern: new RegExp('^proposal_'),
-  unbondingTokenPattern: new RegExp('^unbonding_'),
-  votingReceiptPattern: new RegExp('^voted_on_'),
+export const assetPatterns: AssetPatterns = {
+  lpNft: new RegExp('^lpnft_'),
+  delegationToken: new RegExp('^delegation_'),
+  proposalNft: new RegExp('^proposal_'),
+  unbondingToken: new RegExp('^unbonding_'),
+  votingReceipt: new RegExp('^voted_on_'),
 };
 
-// TODO: This should only have things like icon or other fields not available in assets proto
-export const assets: Asset[] = [
+export interface Asset {
+  metadata: DenomMetadata;
+  icon?: string;
+}
```

Comment:
this should be renamed so it is never confused with the pb Asset type that is used in many places

Reply:
I don't have context here but why do we want this at all? shouldn't the icon be part of the `DenomMetadata`?